buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/YamlRestCompatTestPlugin.java

```diff
@@ -0,0 +1,189 @@
+
+    @Override
+    public void apply(Project project) {
+
+        project.getPluginManager().apply(ElasticsearchJavaPlugin.class);
+        project.getPluginManager().apply(TestClustersPlugin.class);
+        project.getPluginManager().apply(RestTestBasePlugin.class);
+        project.getPluginManager().apply(RestResourcesPlugin.class);
+        project.getPluginManager().apply(YamlRestTestPlugin.class);
+
+        RestResourcesExtension extension = project.getExtensions().getByType(RestResourcesExtension.class);
+
+        // create source set
+        SourceSetContainer sourceSets = project.getExtensions().getByType(SourceSetContainer.class);
+        SourceSet yamlCompatTestSourceSet = sourceSets.create(SOURCE_SET_NAME);
+        SourceSet yamlTestSourceSet = sourceSets.getByName(YamlRestTestPlugin.SOURCE_SET_NAME);
+        GradleUtils.extendSourceSet(project, YamlRestTestPlugin.SOURCE_SET_NAME, SOURCE_SET_NAME);
+
+        // create the test cluster container, and always use the default distribution
+        ElasticsearchCluster testCluster = createTestCluster(project, yamlCompatTestSourceSet);
+        testCluster.setTestDistribution(TestDistribution.DEFAULT);
+
+        // TODO: this is pretty fragile but the existing logic to checkout prior version branches isn't general purpose
+        // eventually we will want to test against multiple minor versions and will need a better way to checkout an arbitrary prior branch
+        // however, for now we will just get a reference to the checkout directory for ":distribution:bwc:minor:checkoutBwcBranch"
+        int priorMajorVersion = VersionProperties.getElasticsearchVersion().getMajor() - 1;
+        final Path checkoutDir = project.findProject(":distribution:bwc:minor")
+            .getBuildDir()
+            .toPath()
+            .resolve("bwc")
+            .resolve("checkout-" + priorMajorVersion + ".x");
+
+        // copy compatible rest specs
+        Configuration compatSpec = project.getConfigurations().create("compatSpec");
+        Configuration xpackCompatSpec = project.getConfigurations().create("xpackCompatSpec");
+        Configuration additionalCompatSpec = project.getConfigurations().create("additionalCompatSpec");
+        Provider<CopyRestApiTask> copyCompatYamlSpecTask = project.getTasks()
+            .register("copyRestApiCompatSpecsTask", CopyRestApiTask.class, task -> {
+                task.includeCore.set(extension.restApi.getIncludeCore());
+                task.includeXpack.set(extension.restApi.getIncludeXpack());
+                task.sourceSetName = SOURCE_SET_NAME;
+                task.skipHasRestTestCheck = true;
+                task.coreConfig = compatSpec;
+                project.getDependencies()
+                    .add(
+                        task.coreConfig.getName(),
+                        project.files(checkoutDir.resolve("rest-api-spec/src/main/resources").resolve(RELATIVE_API_PATH))
```

Comment:
same here. We should not rely on file dependencies and provide these dependencies as artifacts of the bwc project.